nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/AbstractGCPProcessor.java

```diff
@@ -106,6 +106,16 @@
             .sensitive(true)
             .build();
 
+    // https://cloud.google.com/storage/docs/request-endpoints#storage-set-client-endpoint-java
+    public static final PropertyDescriptor STORAGE_API_HOST = new PropertyDescriptor
+            .Builder().name("storage-api-host")
+            .displayName("Storage API Host")
+            .description("Cloud Storage client libraries manage request endpoints automatically. Optionally, you can set the request endpoint manually.")
```

Comment:
Recommend adjusting the description to include some additional details on the default setting and expected custom behavior.
```suggestion
            .description("Overrides the default storage URL. Configuring an alternative Storage API Host URL also overrides the HTTP Host header on requests as described in the Google documentation for Private Service Connections.")
```